File: API/ContaCentral/ContaCentral/Domain/Models/Transacao.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace ContaCentral.Domain.Models
{
    public class Transacao
    {
        public int Id { get; set; }
        public decimal Valor { get; set; }
        public DateTime DataHora { get; set; }
        public string Descricao { get; set;}
        public int OrigemTipo { get; set; }
        public int OrigemId { get; set; }

        public int CarteiraId { get; set; }
        [ForeignKey("CarteiraId")]
        public Carteira Carteira { get; set; }
    }
}

File: Website/src/app/domain/models/transacao.ts
export class Transacao{
    id:number
    valor: number
    dataHora: string
    descricao: string
    origemTipo : number
    origemId : number
    carteiraId : number

    constructor(id:number, valor: number,dataHora: string, descricao: string,  origemTipo : number, origemId : number, carteiraId : number) {
        //specify your own constructor logic
        this.id = id;
        this.valor = valor;
        this.dataHora = dataHora;
        this.descricao = descricao;
        this.origemTipo = origemTipo;
        this.origemId = origemId;
        this.carteiraId = carteiraId;
    }
}

